### PR TITLE
Entity mapping and dependencies config

### PR DESCRIPTION
**Pull request by steliomo**
I had to use hibernate xml mapping instead of annotations. There is a bug on version 1.11.5 in classes 
BaseOpenmrsMetadata and BaseOpenmrsData. They are using @ManyToOne with @Column annotations while should be  @ManyToOne with @JoinColumn. It was fixed on version 1.11.6 and above.

### CODE REVIEW

**Comment by ningosi** [comment]
@steliomo good catch! and is it running on the higher versions than 1.11.5?

**Comment by valvijo** [comment]
Well done @steliomo! Maybe you should let someone else to review and merge your PRs. This will also give the chance for another team member to learn what you did.

File: api/pom.xml [hunk 1]
```diff
@@ -19,21 +19,21 @@
 			<artifactId>fixture-factory</artifactId>
 			<version>3.1.0</version>
 		</dependency>
-		
+
 		<dependency>
```
Comment (ningosi) [inline api/pom.xml:23]: I have always been declaring all the dependencies in the main pom.xml and just reference them in different places including api and omod projects by just writing 2 lines of groupId and ArtifactId tags, other than repeating them in each project. Things that are declraed in the main pom are regarded as global and their references can be accessed anywhere else in the project including the config.xml and moduleApplicationContext.xml. Though, this NOT  a show stopper, but is the BEST practice for openMRS modules.

Comment (steliomo) [reply api/pom.xml:23]: I really agree with you @ningosi and support you idea. I'll do everything I can to follow that approach.

Thanks

**Comment by steliomo** [comment]
@valvijo your are  right. That's our dev process, someone else need to see and merge my commits. Many thanks.

**Pull request merged**